wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/hc/ServersChecker.java

```diff
@@ -195,10 +199,21 @@ ServerChecker doCreateChecker(URL url, String serverRef) {
     // workaround needed because terminal server doesn't have endpoint to check it readiness
     if ("terminal".equals(serverRef)) {
       return new TerminalHttpConnectionServerChecker(
-          url, machineName, serverRef, 3, 180, TimeUnit.SECONDS, timer);
+          url, machineName, serverRef, 3, 180, 1, TimeUnit.SECONDS, timer);
+    }
+    if ("wsagent/http".equals(serverRef)) {
+      return new HttpConnectionServerChecker(
+          url,
+          machineName,
+          serverRef,
+          3,
```

Comment:
@ibuziuk can you elaborate why terminal and exec agents don't need a threshold = 3 whereas wsagent needs it?

Reply:
Sorry, misinterpreted the diff